how-to/how-to-import-kata-logs-with-fluentd.md

```diff
@@ -0,0 +1,403 @@
+
+> **Note:** The below `fluentd` config fragment is in the "older style" to match  the Minikube version of
+> `fluentd`. If using a more up to date version of `fluentd`, you may need to update some parameters, such as
+> using `matches` rather than `filters` and placing `@` before `type`. Your `fluentd` should warn you in its
+> logs if such updates are necessary.
+
+```
+<source>
+  type systemd
+  tag kata-containers
+  path /run/log/journal
+  pos_file /run/log/journal/kata-journald.pos
+  filters [{"SYSLOG_IDENTIFIER": "kata-runtime"}, {"SYSLOG_IDENTIFIER": "kata-proxy"}, {"SYSLOG_IDENTIFIER": "kata-shim"}]
+  read_from_head true
+</source>
+```
+
+We then apply the new YAML, and restart the `fluentd` pod (by killing it, and letting the `ReplicationController`
+start a new instance, which will pick up the new `ConfigurationMap`):
+
+```bash
+$ kubectl apply -f new-fluentd-cm.yaml
+$ kubectl delete pod -n=kube-system fluentd-es-XXXXX
+```
+
+Now open the Kibana UI to the Minikube EFK addon, and launch a Kata QEMU based test pod in order to
+generate some Kata specific log entries:
+
+```bash
+$ minikube addons open efk
+$ cd $GOPATH/src/github.com/kata-containers/packaging/kata-deploy
+$ kubectl apply -f examples/nginx-deployment-qemu.yaml
+```
+
+Looking at the Kibana UI, we can now see that some `kata-runtime` tagged records have appeared:
+
+![Kata tags in EFK](./images/efk_kata_tag.png)
+
+If we now filter on that tag, we can see just the Kata related entries
+
+![Kata tags in EFK](./images/efk_filter_on_tag.png)
+
+If we expand one of those entries, we can see we have imported useful information. You can then
+sub-filter on, for instance, the `SYSLOG_IDENTIFIER` to differentiate the Kata components, and
+on the `PRIORITY` to filter out critical issues etc.
+
+Kata generates a significant amount of Kata specific information, which can be seen as
+[`logfmt`](https://github.com/kata-containers/tests/tree/master/cmd/log-parser#logfile-requirements).
+data contained in the `MESSAGE` field. Imported as-is, there is no easy way to filter on that data
+in Kibana:
+
+![Kata tags in EFK](./images/efk_syslog_entry_detail.png).
+
+We can however furter sub-parse the Kata entries using the
+[fluentd plugins](https://docs.fluentbit.io/manual/parser/logfmt) that will parse
+`logfmt` formatted data. We can utilise these to parse the sub-fields using a fluentd filter
+section. At the same time, we will prefix the new fields with `kata_` to make it clear where
+they have come from:
+
+```
+<filter kata-containers>
+  @type parser
+  key_name MESSAGE
+  format logfmt
+  reserve_data true
+  inject_key_prefix kata_
+</filter>
+```
+
+The Minikube `fluentd` version does not come with the `logfmt` parser installed, so we will run a local
+test to check the parsing works. The resulting output from `fluentd` is:
+
+```json
+2020-02-21 10:31:27.810781647 +0000 kata-containers:
```

Comment:
Nit: this isn't rendering entirely as expected in GitHub as it's not (just) JSON due to the first line, but meh I guess.

Reply:
I know. it's a bit sad. I tried to work out if there was a way to combine two code blocks, one `json` and one not, without getting a gap rendered between them, but I could not. I've dropped the `json` formatting...